flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -453,7 +440,10 @@ public StreamExecutionEnvironment setBufferTimeout(long timeoutMillis) {
             throw new IllegalArgumentException("Timeout of buffer must be non-negative or -1");
         }
 
-        this.bufferTimeout = timeoutMillis;
+        if (timeoutMillis == ExecutionOptions.DISABLED_NETWORK_BUFFER_TIMEOUT) {
+            this.configuration.set(ExecutionOptions.BUFFER_TIMEOUT_ENABLED, false);
+        }
+        this.configuration.set(ExecutionOptions.BUFFER_TIMEOUT, Duration.ofMillis(timeoutMillis));
         return this;
```

Comment:
I'm curious could we deprecate `setBufferTimeout` and `getBufferTimeout` and remove them in 2.0?

I didn't notice any benefit for them, and `DISABLED_NETWORK_BUFFER_TIMEOUT` is more clear for users than -1. WDYT?

Reply:
In fact, the getter/setter API provides a user-friendly configuration method, and for me, I have not yet found a very clear and compelling reason for removing this API. 
WDYT？

Reply:
Thanks for the clarification. It's fine for me.